content/techniques/queues.md

```diff
@@ -6,15 +6,445 @@ Queues are a powerful design pattern that help you deal with common application
+
+@Processor('audio')
+export class AudioConsumer {}
+```
+
+> info **Hint** Consumers must be registered as `providers` so the `@nestjs/bullmq` package can pick them up.
+
+Where the decorator's string argument (e.g., `'audio'`) is the name of the queue to be associated with the class methods.
+
+```typescript
+import { Processor, WorkerHost } from '@nestjs/bullmq';
+import { Job } from 'bullmq';
+
+@Processor('audio')
+export class AudioConsumer extends WorkerHost {
+  async process(job: Job<any, any, string>): Promise<any> {
+    let progress = 0;
+    for (i = 0; i < 100; i++) {
+      await doSomething(job.data);
+      progress += 1;
+      await job.progress(progress);
+    }
+    return {};
+  }
+}
+```
+
+The process method is called whenever the worker is idle and there are jobs to process in the queue. This handler method receives the `job` object as its only argument. The value returned by the handler method is stored in the job object and can be accessed later on, for example in a listener for the completed event.
+
+`Job` objects have multiple methods that allow you to interact with their state. For example, the above code uses the `progress()` method to update the job's progress. See [here](https://api.docs.bullmq.io/classes/v4.Job.html) for the complete `Job` object API reference.
+
+In the older version, Bull, you could designate that a job handler method will handle **only** jobs of a certain type (jobs with a specific `name`) by passing that `name` to the `@Process()` decorator as shown below.
+
+> warning **Warning** This doesn't work with BullMQ, keep reading.
+
+```typescript
+@Process('transcode')
+async transcode(job: Job<unknown>) { ... }
+```
+
+This behavior is not supported in BullMQ due to confusions it generated. Instead, you need switch cases to call different services or logic for each job name:
+
+```typescript
+import { Processor, WorkerHost } from '@nestjs/bullmq';
+import { Job } from 'bullmq';
+
+@Processor('audio')
+export class AudioConsumer extends WorkerHost {
+  async process(job: Job<any, any, string>): Promise<any> {
+    switch (job.name) {
+      case 'transcode': {
+        let progress = 0;
+        for (i = 0; i < 100; i++) {
+          await doSomething(job.data);
+          progress += 1;
+          await job.progress(progress);
+        }
+        return {};
+      }
+      case 'concatenate': {
+        await doSomeLogic2();
+        break;
+      }
+    }
+  }
+}
+```
+
+This is covered in the [named processor](https://docs.bullmq.io/patterns/named-processor) section of the BullMQ documentation.
+
+#### Rate limiting
+
+It is possible to configure the workers so that they obey a given rate limiting option as described in the [Rate Limiting](https://docs.bullmq.io/guide/rate-limiting) section.
+
+[TODO(saratscheff): It seems @nestjs/bullmq hasn't added support for this yet. In Bull it was handled at a queue level. In bullmq it was moved to be handled at a worker level.]
```

Comment:
Not sure if its worth adding this here. Devs should be able to figure out rate limiting based on BullMQ's own docs 👍 .